hw_30_plan/js/script.js

```diff
@@ -0,0 +1,132 @@
+const listItems = [
+    'Work out',
+    'Get some sleep',
+    'Call the boyfriend',
+    'Cook the dinner',
+    'Watch Netflix',
+    'Take a shower',
+    'Walk with the dog',
+]
+
+let ul = document.querySelector('#ul');
+
+let menu = document.querySelector('#menu');
+
+class Menu {
+    constructor(elem) {
+        this._elem = elem;
+        elem.onclick = this.onClick.bind(this);
+    }
+
+    addStart() {
+        let newItem = document.createElement('li');
+        newItem.textContent = listItems[Math.floor(Math.random() * listItems.length)];
+        ul.prepend(newItem);
+    }
+
+    addEnd() {
+        let newItem = document.createElement('li');
+        newItem.textContent = listItems[Math.floor(Math.random() * listItems.length)];
+        ul.append(newItem);
+    }
+
+    delete() {
+        let selectedItems = Array.from(ul.children).filter(item => item.classList.contains('selected'));
+        selectedItems.forEach(item => item.remove());
+    }
+
+    sort() {
+        let selectedItems = Array.from(ul.children).filter(item => item.classList.contains('selected'));
+        selectedItems.reverse().forEach(item => ul.prepend(item));
+    }
+
+    onClick(event) {
+        let action = event.target.dataset.action;
+        if (action) {
+            this[action]();
+        }
+    }
+}
+
+new Menu(menu);
+
+ul.addEventListener('mousedown', function (e) {
+    e.preventDefault();
+})
+
+ul.addEventListener('click', function (e) {
+    if (e.target == this) {
+        return false;
+    }
+
+    if (!e.ctrlKey && !e.metaKey && !e.shiftKey) {
+        clearSelected(this.children);
+        addSelected(e.target);
+    }   
+
+    if (e.ctrlKey || e.metaKey) {
+        toggleSelected(e.target);
+    }
+
+    if (e.shiftKey) {
+        addGroup(e.target, Array.from(this.children));
+    }
+
+})
```

Comment:
You have strange structure in this function. First you have logic of exiting from function - it's ok. But then you have a conditional branching, which really hard to read.
you need to do something like:
```javascript
if (e.ctrlKey || e.metaKey) {
 // ctrlKey||metaKey behaviour 
} else if (e.shiftKey) {
 // shiftKey behaviour
} else {
   // any other behaviour
}
```